pallets/protos/src/lib.rs

```diff
@@ -408,8 +417,15 @@ pub mod pallet {
 			// hash the immutable data, this is also the unique proto id
 			// to compose the V1 Cid add this prefix to the hash: (str "z" (base58
 			// "0x0155a0e40220"))
-			let proto_hash = blake2_256(&data);
-			let data_len = data.len();
+		  // Doesn't work, we need to store on state non Local data...!!
```

Comment:
This is maybe wrong.
Let's simply store the enum as is and add a enum in proto struct to keep track of what it is.
That way even Arweave hash or whatever is fetched thru bitswap.